storagenode/vouchers/vouchers_test.go

```diff
@@ -0,0 +1,204 @@
+		defer ctx.Cleanup()
+
+		vdb := db.Vouchers()
+
+		satellite := testidentity.MustPregeneratedSignedIdentity(0, storj.LatestIDVersion())
+		storagenode := testidentity.MustPregeneratedSignedIdentity(1, storj.LatestIDVersion())
+
+		expiration, err := ptypes.TimestampProto(time.Now().UTC().Add(24 * time.Hour))
+		require.NoError(t, err)
+
+		voucher := &pb.Voucher{
+			SatelliteId:   satellite.ID,
+			StorageNodeId: storagenode.ID,
+			Expiration:    expiration,
+		}
+
+		// GetValid with no entry
+		result, err := vdb.GetValid(ctx, []storj.NodeID{satellite.ID})
+		require.NoError(t, err)
+		assert.Nil(t, result)
+
+		// basic Put test
+		err = vdb.Put(ctx, voucher)
+		require.NoError(t, err)
+
+		// basic NeedVoucher test
+		expirationBuffer := 48 * time.Hour
+		need, err := vdb.NeedVoucher(ctx, satellite.ID, expirationBuffer)
+		require.NoError(t, err)
+		require.True(t, need)
+
+		// basic GetValid test
+		result, err = vdb.GetValid(ctx, []storj.NodeID{satellite.ID})
+		require.NoError(t, err)
+		require.Equal(t, voucher.SatelliteId, result.SatelliteId)
+		require.Equal(t, voucher.StorageNodeId, result.StorageNodeId)
+
+		expectedTime, err := ptypes.Timestamp(voucher.GetExpiration())
+		require.NoError(t, err)
+		actualTime, err := ptypes.Timestamp(result.GetExpiration())
+		require.NoError(t, err)
+
+		require.Equal(t, expectedTime, actualTime)
+
+		// Test duplicate satellite id updates voucher
+		voucher.Expiration, err = ptypes.TimestampProto(time.Now().UTC().Add(48 * time.Hour))
+		require.NoError(t, err)
+
+		err = vdb.Put(ctx, voucher)
+		require.NoError(t, err)
+
+		result, err = vdb.GetValid(ctx, []storj.NodeID{satellite.ID})
+		require.NoError(t, err)
+
+		expectedTime, err = ptypes.Timestamp(voucher.GetExpiration())
+		require.NoError(t, err)
+		actualTime, err = ptypes.Timestamp(result.GetExpiration())
+		require.NoError(t, err)
+
+		require.Equal(t, expectedTime, actualTime)
+	})
+}
+
+func TestVouchersService(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 5, StorageNodeCount: 1, UplinkCount: 0,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		node := planet.StorageNodes[0]
+		node.Vouchers.Loop.Pause()
+
+		// node type needs to be set to receive vouchers
+		for _, sat := range planet.Satellites {
+			_, err := sat.Overlay.Service.UpdateNodeInfo(ctx, node.ID(), &pb.InfoResponse{Type: pb.NodeType_STORAGE})
+			require.NoError(t, err)
+		}
+
+		// assert no vouchers
+		for _, sat := range planet.Satellites {
+			voucher, err := node.DB.Vouchers().GetValid(ctx, []storj.NodeID{sat.ID()})
+			require.NoError(t, err)
+			assert.Nil(t, voucher)
+		}
+
+		// run service and check vouchers have been received
+		err := node.Vouchers.RunOnce(ctx)
+		require.NoError(t, err)
+
+		for _, sat := range planet.Satellites {
+			voucher, err := node.DB.Vouchers().GetValid(ctx, []storj.NodeID{sat.ID()})
+			require.NoError(t, err)
+			assert.NotNil(t, voucher)
+		}
```

Comment:
this is totally unnecessary since your test coverage is already amazing, but just as a fun idea it might be cool to write a test that runs the voucher service when the storagenode is not vetted, then runs it again later and show that it is vetted